test/Integration/IntegrationTestFixture.cs

```diff
@@ -41,8 +80,184 @@ protected void StartAzurite()
+
+            var startInfo = new ProcessStartInfo
+            {
+                // The full path to the Functions CLI is required in the ProcessStartInfo because UseShellExecute is set to false.
+                // We cannot both use shell execute and redirect output at the same time: https://docs.microsoft.com//dotnet/api/system.diagnostics.processstartinfo.redirectstandardoutput#remarks
+                FileName = TestUtils.GetFunctionsCoreToolsPath(),
+                Arguments = $"start --verbose --port {port} {functionsArg}",
+                WorkingDirectory = workingDirectory,
+                WindowStyle = ProcessWindowStyle.Hidden,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                UseShellExecute = false
+            };
+
+            // Always disable telemetry during test runs
+            startInfo.EnvironmentVariables[TelemetryOptoutEnvVar] = "1";
+
+            Console.WriteLine($"Starting {startInfo.FileName} {startInfo.Arguments} in {startInfo.WorkingDirectory}");
+
+            var functionHost = new Process
+            {
+                StartInfo = startInfo
+            };
+
+            this.FunctionHostList.Add(functionHost);
+
+            // Register all handlers before starting the functions host process.
+            var taskCompletionSource = new TaskCompletionSource<bool>();
+            void SignalStartupHandler(object sender, DataReceivedEventArgs e)
+            {
+                // This string is printed after the function host is started up - use this to ensure that we wait long enough
+                // since sometimes the host can take a little while to fully start up
+                if (e.Data?.Contains(" Host initialized ") == true)
+                {
+                    taskCompletionSource.SetResult(true);
+                }
+            };
+            functionHost.OutputDataReceived += SignalStartupHandler;
+
+            functionHost.Start();
+            functionHost.OutputDataReceived += this.GetTestOutputHandler(functionHost.Id);
+            functionHost.ErrorDataReceived += this.GetTestOutputHandler(functionHost.Id);
+            functionHost.BeginOutputReadLine();
+            functionHost.BeginErrorReadLine();
+
+            Console.WriteLine($"Waiting for Azure Function host in {workingDirectory} to start...");
+
+            const int FunctionHostStartupTimeoutInSeconds = 60;
+            bool isCompleted = taskCompletionSource.Task.Wait(TimeSpan.FromSeconds(FunctionHostStartupTimeoutInSeconds));
+            Assert.True(isCompleted, $"Functions host in {workingDirectory} did not start within specified time.");
+
+            // Give additional time to Functions host to setup routes for the HTTP triggers so that the HTTP requests
+            // made from the test methods do not get refused.
+            const int BufferTimeInSeconds = 5;
```

Comment:
nit: this would ideally live in a TestConstants.cs file or something of that nature